kube-controllers/tests/fv/metrics_test.go

```diff
@@ -124,192 +138,254 @@ var _ = Describe("kube-controllers metrics tests", func() {
+				`ipam_ippool_size{ippool="test-ippool-3"} 256`,
 			},
-			metav1.CreateOptions{})
-		Expect(err).NotTo(HaveOccurred())
-		_, err = k8sClient.CoreV1().Nodes().Create(context.Background(),
-			&v1.Node{
-				TypeMeta:   metav1.TypeMeta{Kind: "Node", APIVersion: "v1"},
-				ObjectMeta: metav1.ObjectMeta{Name: nodeB},
-				Spec:       v1.NodeSpec{},
-			},
-			metav1.CreateOptions{})
-		Expect(err).NotTo(HaveOccurred())
-		_, err = k8sClient.CoreV1().Nodes().Create(context.Background(),
-			&v1.Node{
-				TypeMeta:   metav1.TypeMeta{Kind: "Node", APIVersion: "v1"},
-				ObjectMeta: metav1.ObjectMeta{Name: nodeC},
-				Spec:       v1.NodeSpec{},
+			[]string{
+				`ipam_allocations_`,
+				`ipam_blocks_`,
+				`ipam_gc_`,
 			},
-			metav1.CreateOptions{})
-		Expect(err).NotTo(HaveOccurred())
-
-		// Assert no IPAM metrics reported yet.
-		var out string
-		Eventually(func() error {
-			out, err = getMetrics(fmt.Sprintf("http://%s:9094/metrics", kubeControllers.IP))
-			return err
-		}, 10*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
-		Expect(out).NotTo(ContainSubstring("ipam_"))
+			kubeControllers.IP,
+			10*time.Second, 1*time.Second,
+		)
 
-		// Allocate a pod IP address and thus a block and affinity to NodeA.
+		// Allocate pod IP addresses from pool 1 and 2, and thus blocks and affinities to NodeA.
 		handleA := "handleA"
-		attrs := map[string]string{"node": nodeA, "pod": "pod-a", "namespace": "default"}
-		err = calicoClient.IPAM().AssignIP(context.Background(), ipam.AssignIPArgs{
-			IP: net.MustParseIP("192.168.0.1"), HandleID: &handleA, Attrs: attrs, Hostname: nodeA,
-		})
-		Expect(err).NotTo(HaveOccurred())
+		handleA2 := "handleA2"
+		createPod("pod-a", "192.168.0.1", handleA, nodeA, k8sClient, calicoClient)
+		createPod("pod-a2", "172.16.0.1", handleA2, nodeA, k8sClient, calicoClient)
 
 		// Allocate an IPIP, VXLAN and WG address to NodeA as well.
 		handleAIPIP := "handleAIPIP"
-		attrs = map[string]string{"node": nodeA, "type": "ipipTunnelAddress"}
-		err = calicoClient.IPAM().AssignIP(context.Background(), ipam.AssignIPArgs{
-			IP: net.MustParseIP("192.168.0.2"), HandleID: &handleAIPIP, Attrs: attrs, Hostname: nodeA,
-		})
-		Expect(err).NotTo(HaveOccurred())
-
 		handleAVXLAN := "handleAVXLAN"
-		attrs = map[string]string{"node": nodeA, "type": "vxlanTunnelAddress"}
-		err = calicoClient.IPAM().AssignIP(context.Background(), ipam.AssignIPArgs{
-			IP: net.MustParseIP("192.168.0.3"), HandleID: &handleAVXLAN, Attrs: attrs, Hostname: nodeA,
-		})
-		Expect(err).NotTo(HaveOccurred())
-
 		handleAWG := "handleAWireguard"
-		attrs = map[string]string{"node": nodeA, "type": "wireguardTunnelAddress"}
-		err = calicoClient.IPAM().AssignIP(context.Background(), ipam.AssignIPArgs{
-			IP: net.MustParseIP("192.168.0.4"), HandleID: &handleAWG, Attrs: attrs, Hostname: nodeA,
-		})
-		Expect(err).NotTo(HaveOccurred())
+		allocateInterfaceIpWithHandle("192.168.0.2", handleAIPIP, nodeA, "ipipTunnelAddress", calicoClient)
```

Comment:
nit: `InterfaceIP` instead of `InterfaceIp` - golang conventions say to use same-case for acronyms